src/clients/Device.js

```diff
@@ -723,39 +541,68 @@ class Device extends EventEmitter {
     return Messages.getResponseType(request);
   };
 
+  // todo make return type annotation
+  getDescription = async (): Promise<*> => {
+    const isBusy = !this._isSocketAvailable(null);
+    if (isBusy) {
+      throw new Error('This core is locked during the flashing process.');
+    }
+
+    try {
+      await this._ensureWeHaveIntrospectionData();
+
+      return {
+        firmware_version: this._productFirmwareVersion,
+        product_id: this._particleProductId,
+        state: this._deviceFunctionState,
+      };
+    } catch (error) {
+      throw new Error('No device state!');
+    }
+  };
+
   /**
    * Ensures we have introspection data from the core, and then
    * requests a variable value to be sent, when received it transforms
    * the response into the appropriate type
-   * @param name
-   * @param type
-   * @param callback - expects (value, buf, err)
-   */
-  _getVariable = async (
+   **/
+  getVariableValue = async (
     name: string,
-    type: string,
   ): Promise<*> => {
+    const isBusy = !this._isSocketAvailable(null);
+    if (isBusy) {
+      throw new Error('This core is locked during the flashing process.');
+    }
+
     await this._ensureWeHaveIntrospectionData();
     if (!this._hasParticleVariable(name)) {
       throw new Error('Variable not found');
     }
 
     const messageToken = this.sendMessage(
-      'VariableRequest', { name },
+      'VariableRequest',
+      { name },
     );
     const message = await this.listenFor(
       'VariableValue',
       null,
       messageToken,
     );
+
     return this._transformVariableResult(name, message);
   };
 
-  _setVariable = async (
+  // TODO refactor, make sure if we need this at all
+  setVariableValue = async (
```

Comment:
not sure if we need this at all...now we use it `nowhere` in the codebase..and I'm not sure if this works at all. Generally, if it's completely written it'd be different from `getVariableValue` only by putting payload as 3rd arg here:
const token = this.sendMessage('VariableRequest', { name }, payload);